pkg/api/annotations.go

```diff
@@ -54,17 +54,7 @@ func (hs *HTTPServer) GetAnnotations(c *contextmodel.ReqContext) response.Respon
 		query.Limit = defaultAnnotationsLimit
 	}
 
-	// When dashboard UID present in the request, we ignore dashboard ID
-	if query.DashboardUID != "" {
-		dq := dashboards.GetDashboardQuery{UID: query.DashboardUID, OrgID: c.GetOrgID()}
-		dqResult, err := hs.DashboardService.GetDashboard(c.Req.Context(), &dq)
-		if err != nil {
-			return response.Error(http.StatusBadRequest, "Invalid dashboard UID in annotation request", err)
-		} else {
-			query.DashboardID = dqResult.ID // nolint:staticcheck
```

Comment:
no need to set the internal id -- the queries are all by UID and we are saving by UID so i think this falls safely into the zone of deprecated for a long while and no longer need to support